android-components/components/feature/session/src/main/java/mozilla/components/feature/session/middleware/LastAccessMiddleware.kt

```diff
@@ -40,8 +40,7 @@ class LastAccessMiddleware : Middleware<BrowserState, BrowserAction> {
         when (action) {
             is TabListAction.RemoveTabAction,
             is TabListAction.RemoveTabsAction,
-            // NB: RemoveAllNormalTabsAction never updates tab selection
-            is TabListAction.RemoveAllPrivateTabsAction,
+            // NB: RemoveAllNormalTabsAction and RemoveAllPrivateTabsAction never updates tab selection
```

Comment:
This was to fix another failing test. If we are changing the behavior such that a tab doesn't get selected, we shouldn't need to worry about the `lastAccess` property getting updated at that time.

Reply:
Do we need to also remove `RemoveAllPrivateTabsAction` handling from `selectionBeforeRemoval ` on [L31](https://github.com/mozilla-mobile/firefox-android/pull/5284/files#diff-dbe0c53ebd3a3324ce877e2e00949aa033ae930d30309d3ddeb55fd641d9f355L31)?

I believe so because `RemoveAllPrivateTabsAction` won't update the tab selection now and `selectionBeforeRemoval` is only ever used in this conditional block.

Reply:
It wasn't technically necessary because `newSelection` in the below conditional will now be null in the `RemoveAllPrivateTabsAction` case, but I agree that we don't want it lingering as tech debt to cause potential bugs. Thanks for catching that!